helm/nginx-ingress-controller-app/tests/abs/test_basic_cluster.py

```diff
@@ -188,3 +118,43 @@ def test_ingress_creation(
         time.sleep(5)
 
     assert last_status == 200
+
+
+@pytest.mark.functional
+@pytest.mark.flaky(reruns=5, reruns_delay=10)
+def test_multiple_ingress_controllers(
+    kube_cluster: Cluster, ic_deployment: List[pykube.Deployment], chart_version
+):
+    logger.info("applying manifests")
+    # apply test manifests
+    kube_cluster.kubectl(
+        "apply", filename="multi-controller-manifests.yaml", output_format=""
+    )
+
+    # shortcut function with fixed namespace
+    # (defined in multi-controller-manifests.yaml)
+    kubectl = partial(
+        kube_cluster.kubectl, namespace="second-ingress-controller", output_format=""
+    )
+    logger.info("patching app with current chart version")
+    # patch the app cr with the right version
+    patch = dumps([{"op": "replace", "path": "/spec/version", "value": chart_version}])
+    kubectl(f"patch app second-ingress-controller --type=json", patch=patch)
+
+    # just wait a little for app-operator to create the deployment (so kubectl wait does not fail)
+    time.sleep(10)
```

Comment:
You could do something like a `kubectl get pods --select [..]` until something shows up instead of sleeping. Makes the test more robust for edge cases and faster in general. But just as a note, maybe not too important.